kubernetes/provider.go

```diff
@@ -364,3 +365,49 @@ func initializeConfiguration(d *schema.ResourceData) (*restclient.Config, error)
 	log.Printf("[INFO] Successfully initialized config")
 	return cfg, nil
 }
+
+var useadmissionregistrationv1beta1 *bool
+
+func useAdmissionregistrationV1beta1(conn *kubernetes.Clientset) (bool, error) {
+	if useadmissionregistrationv1beta1 != nil {
+		return *useadmissionregistrationv1beta1, nil
+	}
+
+	d := conn.Discovery()
+	groups, err := d.ServerGroups()
+	if err != nil {
+		return false, err
+	}
+
+	apiGroup := "admissionregistration.k8s.io"
+	use, err := func() (bool, error) {
+		for _, g := range groups.Groups {
+			if g.Name == apiGroup {
+				preferredVersion := g.PreferredVersion.Version
+				if preferredVersion == "v1" {
+					return false, nil
+				} else if preferredVersion == "v1beta1" {
+					return true, nil
+				} else {
+					// see if v1 or v1beta are there even if they are not preferred
+					for _, v := range g.Versions {
+						if v.Version == "v1" {
+							return false, nil
+						} else if v.Version == "v1beta1" {
```

Comment:
Just returning the result of this comparison alone should be enough to establish the truth value.